### ISSUE TITLE
About MODBUS error code

### ISSUE THREAD

**Issue by lnsane-G** [emelianov/modbus-esp8266#59]
Hello, sir. I need some help.
My device is esp32 and uses the 3.0.1 library.
In the actual test, the library reported an error: MODBUS result: E4
I really can't find the reason and ask for your help.
Here is a screenshot of my test code and software.

`
bool cb(Modbus::ResultCode event, uint16_t transactionId, void* data) { // Modbus Transaction callback
  if (event != Modbus::EX_SUCCESS)                  // If transaction got an error
    Serial.printf("Modbus result: %02X\n", event);  // Display Modbus error code
  // if (event == Modbus::EX_TIMEOUT) {    // If Transaction timeout took place
  //   mb.disconnect(remote);              // Close connection to slave and
  //   mb.dropTransactions();              // Cancel all waiting transactions
  // }
  return true;
}

void setup() {
  Serial.begin(9600);
  WiFi.begin("BYJD","BYJD123456");
  
  while (WiFi.status() != WL_CONNECTED) {
    delay(500);
    Serial.print(".");
  }
  Serial.println("");
  Serial.println("WiFi connected");  
  Serial.println("IP address: ");
  Serial.println(WiFi.localIP());
  mb.client();
}

uint16_t v[1]={true};
void loop() {
    if (!mb.isConnected(remote)) {   // Check if connection to Modbus Slave is established
        mb.connect(remote);           // Try to connect if no connection
        Serial.print(".");
    }
    mb.readHreg(remote,0,v,1,cb,1);
    //Serial1.println(v[0]);
    //mb.readCoil(remote,0,v,1,cb,1);
    mb.task(); // Modbus task
    delay(1000); // Pushing interval
}
`
![捕获](https://user-images.githubusercontent.com/48591117/90331148-f68a0780-dfe4-11ea-9c8a-8bd054b970e1.PNG)

**Comment by lnsane-G** [comment]
![enclosure](https://user-images.githubusercontent.com/48591117/90331167-128da900-dfe5-11ea-9fea-0a2dffb8682c.PNG)

**Comment by emelianov** [comment]
Hi,
For the first decrease `delay(1000);` to say `delay(500);`

**Comment by lnsane-G** [comment]
I don't seem to understand

**Comment by lnsane-G** [comment]
> 嗨，
> 第一次减少`delay(1000);`说`delay(500);`

I don't seem to understand

**Comment by emelianov** [comment]
Replace `delay(1000);' with `delay(500);` in your code.
```c
void loop() {
if (!mb.isConnected(remote)) { // Check if connection to Modbus Slave is established
mb.connect(remote); // Try to connect if no connection
Serial.print(".");
}
mb.readHreg(remote,0,v,1,cb,1);
//Serial1.println(v[0]);
//mb.readCoil(remote,0,v,1,cb,1);
mb.task(); // Modbus task
delay(500); // Pushing interval
}
```

**Comment by lnsane-G** [comment]
> Replace`delay(1000);`with`delay(500);`in your code.
> 
> ```c
> void loop() {
> if (!mb.isConnected(remote)) { //Check if connection to Modbus Slave is established
> mb.connect(remote); //Try to connect if no connection
> Serial.print(`.`);
> }
> mb.readHreg(remote,0,v,1,cb,1);
> //Serial1.println(v[0]);
> //mb.readCoil(remote,0,v,1,cb,1);
> mb.task(); //Modbus task
> delay(500); //Pushing interval
> }
> ```
Thank you, but the result is the same
![image](https://user-images.githubusercontent.com/48591117/90331908-e07f4580-dfea-11ea-9b73-7bf689bc861f.png)

**Comment by emelianov** [comment]
E4 is Timeout error. For TCP possible reason is communication error or firewall blocking.

**Comment by lnsane-G** [comment]
> E4 is Timeout error. For TCP possible reason is communication error or firewall blocking.
I changed the router and it still didn't work. But Android Modbus TCP debugging tool can be successfully accessed.
![image](https://user-images.githubusercontent.com/48591117/90332571-7e294380-dff0-11ea-8777-2a91fbe4df61.png)

**Comment by emelianov** [comment]
It was my fault. Fixed in [3.0.2](https://github.com/emelianov/modbus-esp8266/releases/tag/3.0.2)

**Comment by lnsane-G** [comment]
> It was my fault. Fixed in [3.0.2](https://github.com/emelianov/modbus-esp8266/releases/tag/3.0.2)

Thank you very much. I'll test it!

**Comment by lnsane-G** [comment]
Test results are successful, thank you!
![image](https://user-images.githubusercontent.com/48591117/90336700-14b92d00-e010-11ea-8459-7e3c8945fefd.png)
